flocker/ca/_ca.py

```diff
@@ -231,14 +231,81 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True)
+    username = field(mandatory=True, initial=None)
+
+    @classmethod
+    def from_path(cls, path, username):
+        """
+        Load a node certificate from a specified path.
+        """
+        key_filename = b"{user}.key".format(user=username)
+        cert_filename = b"{user}.crt".format(user=username)
```

Comment:
:question:

`username + b".key"` works just as well and without the confusing mix of bytes with new-style formatting (which .. works .. but .. it's sort of gross).